cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -681,6 +681,28 @@ func getLastBumpCommit(gerritAuthor, commitTag string) (string, error) {
 	return outBuf.String(), nil
 }
 
+// getDefaultBranch retrieves the default branch name of a given repository
+// using the provided GitHub client. It fetches the repository details and
+// extracts the default branch. It returns a string representing the default
+// branch name and error if the repository details cannot be retrieved.
+func getDefaultBranch(gc github.Client, org, repo string) (string, error) {
+	repository, err := gc.GetRepo(org, repo)
+	if err != nil {
+		return "", fmt.Errorf("failed to get repository details for %s/%s: %w", org, repo, err)
+	}
+	return repository.DefaultBranch, nil
+}
```

Comment:
TBH adding this wrapper does not simplify the logic much, and the indirection just to extract a single member from the `repository` seems like an overkill.

The callsite in `createOrUpdatePROnDefaultBranch` will look basically identical if it uses `GetRepo` directly, and then just `repository.DefaultBranch` where it uses `defaultBranch` now. I'd prefer that.